File: 0203.js
function longestCommonPrefix(strs) {
    if (strs.length === 0) {
        return "";
    }

    let commonPrefix = "";
    const firstStr = strs[0];

    for (let i = 0; i < firstStr.length; i++) {
        const char = firstStr[i];
        if (!strs.every(str => str[i] === char)) {
            break;
        }
        commonPrefix += char;
    }

    return commonPrefix;
}

// 測試
console.log(longestCommonPrefix(["flower", "flow", "flight"]));  // 輸出: "fl"
console.log(longestCommonPrefix(["dog", "racecar", "car"]));      // 輸出: ""

File: 0203.rb
def longest_common_prefix(strs)
  return "" if strs.empty?

  common_prefix = ""
  first_str = strs.first

  first_str.chars.each_with_index do |char, i|
    break unless strs.all? { |str| str[i] == char }
    common_prefix += char
  end

  common_prefix
end

# 測試案例
puts longest_common_prefix(["flower", "flow", "flight"])  # 輸出: "fl"
puts longest_common_prefix(["dog", "racecar", "car"])      # 輸出: ""
